src/components/rating/rating.tsx

```diff
@@ -131,100 +179,93 @@ export class Rating
-              (i === 1 || i === this.value) && (this.inputFocusRef = el as HTMLInputElement)
-            }
-            type="radio"
-            value={i}
-          />
-        </span>
-      );
-    });
-  }
-
   render() {
-    const { disabled, intlRating, showChip, scale, count, average } = this;
-
     return (
-      <Fragment>
-        <fieldset
-          class="fieldset"
-          disabled={disabled}
-          onBlur={() => (this.hoverValue = null)}
-          onPointerLeave={() => (this.hoverValue = null)}
-          onTouchEnd={() => (this.hoverValue = null)}
-        >
-          <legend class="visually-hidden">{intlRating}</legend>
-          {this.renderStars()}
-        </fieldset>
-        {(count || average) && showChip ? (
-          <calcite-chip scale={scale} value={count?.toString()}>
-            {!!average && <span class="number--average">{average.toString()}</span>}
-            {!!count && <span class="number--count">({count?.toString()})</span>}
-          </calcite-chip>
-        ) : null}
-        <HiddenFormInputSlot component={this} />
-      </Fragment>
+      <Host
+        onBlur={this.handleRatingFocusLeave}
+        onFocus={this.handleRatingFocusIn}
+        onKeyDown={this.handleHostKeyDown}
+        onPointerOut={this.handleRatingPointerOut}
+        onPointerOver={this.handleRatingPointerOver}
+      >
+        <span class="wrapper">
+          <fieldset class="fieldset" disabled={this.disabled}>
+            <legend class="visually-hidden">{this.intlRating}</legend>
+            {this.starsMap.map(
+              ({
+                average,
+                checked,
+                focused,
+                fraction,
+                hovered,
+                id,
+                idx,
+                partial,
+                selected,
+                value
+              }) => {
+                return (
+                  <label
+                    class={{
+                      star: true,
+                      focused,
+                      selected,
+                      hovered,
+                      average,
+                      partial
+                    }}
+                    htmlFor={id}
+                    onPointerDown={this.handleLabelPointerDown}
+                    onPointerOver={this.handleLabelPointerOver}
+                  >
+                    <input
+                      checked={checked}
+                      class="visually-hidden"
+                      disabled={this.disabled || this.readOnly}
+                      id={id}
+                      name={this.guid}
+                      onChange={this.handleInputChange}
+                      onKeyDown={this.handleInputKeyDown}
+                      ref={(el) => {
+                        this.inputRefs[idx] = el;
+                        return (
+                          (value === 1 || value === this.value) &&
+                          (this.inputFocusRef = el as HTMLInputElement)
```

Comment:
Do we still need `inputFocusRef` since we have a reference to all inputs now? Can we use `inputRefs[this.value]` wherever we were previously using `inputFocusRef`?

Reply:
inputFocusRef is still potentially important for react consumers?
I'm hesitant to remove it as it's legacy and I don't fully grok all the "in-the-wild" use cases yet

Reply:
@alisonailea what do you mean react consumers? This is an internal property.

Reply:
React consumers using `ref` would obtain an instance to the `CalciteRating` element, not the internal `<input>` element inside the shadowRoot.  They should be able to do anything they need to do with that instance since the public API for the component would be exposed through that ref.